### PR TITLE
Evaluator library.

### PR DESCRIPTION
**Pull request by putivsky**
Summary: Added to the training framework evaluator library to perfom module evaluation (accuracy) on a fly.

Differential Revision: D17681152

### CODE REVIEW

**Comment by mortzur** [comment]
Sounds awesome!
Yuri, can you please explain a bit about the components that do the evaluation and the other changes you needed to do to make this work?
I mean:
* There is a new function TorchGlowTraining::evaluate() - what does it do? (i.e. does it infer a trained model using known input vs golden accuracy?)
* There are a bunch of other changes in ONNX loader, insertTensor, AvgPool , constants VS placeholders - can you explain why these are necessary for evaluation?

**Comment by putivsky** [comment]
TorchGlowTraining::evaluate() is a design when caller can get a model ready for inference/evaluation.
Other changes are the result of comprehensive testing and fixing bugs, all of them are related to the Standalone training.

**Comment by putivsky** [comment]
Components:
**Provider** goes to the source of training/testing examples and labels and brings back them into format compatible with Glow tensors of specific shapes.
**Loader** loads model from file (currently PyTorch and ONNX are supported, can be extended to Caffe2).
**Trainer** library uses loaded model into Glow Function and TrainingPreparation API to prepare model for training (replacing constants with placeholders, randomize weights, etc.).
**Writer** library to save Glow Function Graph back to ONNX file, compatible with ONNX Loader format.

**Comment by putivsky** [comment]
mor, the most of the code in fb phabricator, would you mind to review PR there?
